metaflow/plugins/conda/conda_step_decorator.py

```diff
@@ -250,30 +251,48 @@ def runtime_init(self, flow, graph, package, run_id):
         self.addl_paths = None
         os.symlink(path_to_metaflow, os.path.join(self.metaflow_home, "metaflow"))
 
-        # Also symlink the INFO version to properly propagate down version information
-        # from, for example, a step-function execution
+        # Symlink the INFO file as well to properly propagate down the Metaflow version
+        # if launching Titus from a meson run for example
         if os.path.isfile(path_to_info):
             os.symlink(path_to_info, os.path.join(self.metaflow_home, "INFO"))
-
-        # Do the same for metaflow_extensions
+        else:
+            # If there is no "INFO" file, we will actually create one in this new
+            # place because we won't be able to properly resolve the EXT_PKG extension
+            # since they may all be linked
```

Comment:
I'll clarify; it refers to what is happening around line 286 in this file. The logic is that without the INFO file, we look around for the metaflow_extensions packages whereas when there is an INFO file, we use the information from there to figure out thee metaflow_extensions packages. In the case of a conda environment, we don't want to "look around" because now things are all installed as part of a custom path (and not distributions) which means we wouldn't be able to do the analysis we do on them (ordering particularly) so we therefore create the INFO file so that we can use it directly in the Conda environment.